### ISSUE TITLE
<targetClasses> configuration isn't working in multi module project

### ISSUE THREAD

**Issue by sushantchoudhary** [hcoles/pitest#512]
I am trying to run pitest on a multi module project. My maven config looks like this

```
<build>
        <pluginManagement>
        </pluginManagement>
        <plugins>
            <plugin>
                <groupId>org.pitest</groupId>
                <artifactId>pitest-maven</artifactId>
                <version>1.4.2</version>
                <configuration>
                    <targetClasses>
                        <param>com.xyz.services.rest.*</param>
                    </targetClasses>
                </configuration>
            </plugin>
        </plugins>
    </build>
```
I have added this to a parent module pom.xml  which is of POM packaging type. I am running `mvn  org.pitest:pitest-maven:mutationCoverage` from CLI, however all the child modules and their dependencies are being mutated instead of just the module with the above package `com.xyz.services.rest.* `.  Could you please suggest if I am missing something? 

I also tested this using https://github.com/STAMP-project/pitmp-maven-plugin plugin but the same results. I can apply filters by module but not by classes.

**Comment by maxgabut** [comment]
I tried to replicate, but things looked fine to me.

However, if you want to skip entire modules, I don't think `targetClasses` is the way to go. As far as I know, maven _decides_ in which module the plugin should run and `targetClasses` only act as a filter for classes to consider. To entirely skip pitest execution in some modules, I'd rather use maven profiles.

**Comment by maxgabut** [comment]
Did my last comment helped you ?
If so, you can close the issue.
If not, you might try to give us more information. Ideally, you could publish a project demonstrating the problem, but it is not always easy if you're working on closed source code. The log of the execution should be much more easy to produce and could also help.

**Comment by sushantchoudhary** [comment]
Thanks @maxgabut , I think I can work it out using pitmp with one of <targetModule> bug fix going out in 1.3.5. Closing this issue for now.
